### ISSUE TITLE
Datagrid: Previously focused date cell is shown after sorting

### ISSUE THREAD

**Issue by s-werking** [infor-design/enterprise#6851]
**Describe the bug**
When a date cell is selected on an editable table, the date column gets sorted, and a new cell in that column gets selected, the previously selected date is shown again.

**To Reproduce**
1. Go to http://localhost:4000/components/datagrid/example-editable.html
2. Click on a date cell
3. Select the date column header to sort
4. Click on any date cell
5. See that the previous date is shown again

**Expected behavior**
It is expected that the dates do not change when a date cell is selected.

**Version**
- ids-enterprise: v4.68.0-dev

**Screenshots**
![previous date is shown](https://user-images.githubusercontent.com/76002134/191971285-964779d1-7810-430e-a405-dc8d712d6972.gif)

**Platform**
 - Infor Application/Team Name: lawson
 - OS Version: Windows11
 - Browser Name: chrome
 - Browser Version: 105.0.5195.127 (Official Build) (64-bit)

**Additional context**
Is it possible to fit this in the 4.68 release?

**Comment by tmcconechy** [comment]
Having a hard time seeing what the bug is here. I tried to edit them all and made them 1-5 and its working correctly but the date values are confusing.

Seems like if i
- click in the field
- sort the date column
- set the sort back too row id like initially

Then it is working. Its sort of like all thats happening is the sorting changes the order.

**Comment by s-werking** [comment]
Yeah, I see that the gif is hard to follow. I can go through the steps to reproduce with the table-specific data to help:

2. Click on a date cell (4/3/2015)
3. Select the date column header to sort (Now the cell that was previously selected is sorted correctly and has the value 7/9/2014)
4. Click on any date cell (when this happens the correct date 7/9/2014 is now changed to the previously selected date before the sort, 4/3/2015, so now the date on that row is changed without actually making a change)

**Comment by tmcconechy** [comment]
Oh I see - and this happens on your side too (not just a sample data issue).

**Comment by s-werking** [comment]
Yeah, this happens for us too

**Comment by s-werking** [comment]
Another scenario that could possibly be caused by the same bug happens when you focus a date cell, reorder the columns, then focus a new date cell. The original cell that was focused is attempted to be filled with the original date regardless of the new column's formatter.

Steps to reproduce:
1. Add columnReorder: true to http://localhost:4000/components/datagrid/example-editable.html
2. Go to http://localhost:4000/components/datagrid/example-editable.html
3. Click on a date cell (4/3/2015)
4. Reorder the columns (Move the date column to the left)
5. Click on any date cell (7/3/2014)
6. See that the previous date is attempted to be filled in the originally focused cell (4.000 which is entered as 4/3/2015)

Gif for reference:
![Previous date cell on new column](https://user-images.githubusercontent.com/76002134/192034607-ee659ca4-bf42-4f36-9058-203b62309c8d.gif)

**Comment by glenlieorillo** [comment]
QA tested and passed.

Windows 11 - Chrome (BrowserStack)
https://main-enterprise.demo.design.infor.com/components/datagrid/example-editable.html

https://user-images.githubusercontent.com/102151352/196414102-9a872df8-c292-433d-9101-7db401918811.mov

Mac Chrome
https://main-enterprise.demo.design.infor.com/components/datagrid/example-editable.html

https://user-images.githubusercontent.com/102151352/196414413-1a0b72ad-3c14-4916-bbcb-5f323c3d6062.mov

http://localhost:4000/components/datagrid/example-editable.html

https://user-images.githubusercontent.com/102151352/196414506-153e6f8b-ecf1-49d8-b21a-5fa81b015bfe.mov

**Comment by s-werking** [comment]
Hi, just a note that this one may not be deployed to main. I'm seeing the issue still on https://main-enterprise.demo.design.infor.com/components/datagrid/example-editable.html but not on http://localhost:4000/components/datagrid/example-editable.html. Thanks

**Comment by glenlieorillo** [comment]
It is now working as expected in the main-enterprise and local host.


https://user-images.githubusercontent.com/102151352/199466711-6687fd39-1c21-4e5e-8c1d-e007ca61a716.mov
